### PR TITLE
introduce decorators for `tm_a_pca`

### PR DESCRIPTION
**Pull request by m7pr**
Part of https://github.com/insightsengineering/teal/issues/1370

<details><summary> Working Example </summary>

```r

devtools::load_all("../teal")
devtools::load_all(".")
plot_title <- teal_transform_module(
  server = make_teal_transform_server(expression({
    plot <- plot + ggtitle("Custom title")
  }))
)

# general data example
data <- teal_data()
data <- within(data, {
  require(nestcolor)
  USArrests <- USArrests
})

app <- init(
  data = data,
  modules = modules(
    tm_a_pca(
      "PCA",
      dat = data_extract_spec(
        dataname = "USArrests",
        select = select_spec(
          choices = variable_choices(
            data = data[["USArrests"]], c("Murder", "Assault", "UrbanPop", "Rape")
          ),
          selected = c("Murder", "Assault"),
          multiple = TRUE
        ),
        filter = NULL
      ),
      decorators = list(plot_title)
    )
  )
)
if (interactive()) {
  shinyApp(app$ui, app$server)
}

```

</details>

### CODE REVIEW

File: R/tm_a_pca.R [hunk 1]
```diff
@@ -570,7 +586,7 @@ srv_a_pca <- function(id, data, reporter, filter_panel_api, dat, plot_height, pl
               ggthemes +
               themes
 
-            print(g)
+            print(plot)
```
Comment (averissimo) [inline R/tm_a_pca.R:589]: This needs to be evaluated after the "decoration", otherwise the code looks wrong:

**Review by averissimo — CHANGES_REQUESTED** [review]

File: R/tm_a_pca.R (hunk 1 above)
Comment (m7pr) [reply R/tm_a_pca.R:589]: good catch

File: R/tm_a_pca.R [hunk 2]
```diff
@@ -224,7 +239,8 @@ ui_a_pca <- function(id, ...) {
               label = "Plot type",
               choices = args$plot_choices,
               selected = args$plot_choices[1]
-            )
+            ),
+            ui_teal_transform_data(ns("decorate"), transformators = args$decorators)
```
Comment (averissimo) [inline R/tm_a_pca.R:243]: This makes more sense in line 261, at the top of the panel labelled `title = "Selected plot specific settings",`

**Review by averissimo — APPROVED** [review]
Problem solved, 1 suggestion

File: R/tm_a_pca.R (hunk 2 above)
Comment (averissimo) [reply R/tm_a_pca.R:243]: Example app:

```r
# PCA

devtools::load_all("../teal")
devtools::load_all(".")
plot_title <- teal_transform_module(
  server = make_teal_transform_server(expression({
    plot <- plot + ggtitle("Custom title")
  }))
)

interactive_decorator <- teal_transform_module(
  ui = function(id) {
    ns <- NS(id)
    div(
      textInput(ns("x_axis_title"), "X axis title", value = "x axis")
    )
  },
  server = function(id, data) {
    moduleServer(id, function(input, output, session) {
      reactive({
        req(data())
        within(data(),
               {
                 plot <- plot +
                   xlab(my_title)
               },
               my_title = input$x_axis_title
        )
      })
    })
  }
)

# general data example
data <- teal_data()
data <- within(data, {
  require(nestcolor)
  USArrests <- USArrests
})

app <- init(
  data = data,
  modules = modules(
    tm_a_pca(
      "PCA",
      dat = data_extract_spec(
        dataname = "USArrests",
        select = select_spec(
          choices = variable_choices(
            data = data[["USArrests"]], c("Murder", "Assault", "UrbanPop", "Rape")
          ),
          selected = c("Murder", "Assault"),
          multiple = TRUE
        ),
        filter = NULL
      ),
      decorators = list(plot_title, interactive_decorator)
    )
  )
)
if (interactive()) {
  shinyApp(app$ui, app$server)
}
```

Comment (averissimo) [reply R/tm_a_pca.R:243]: image

Comment (m7pr) [reply R/tm_a_pca.R:243]: thanks @averissimo - I don't want to hide decorators UI for now and this seems like a bit hidden

**Pull request merged**